examples/eight_schools/main_mcmc.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import, division, print_function
+
+import argparse
+import logging
+import sys
+
+import numpy as np
+import pandas as pd
+import torch
+
+import data
+import pyro
+import pyro.distributions as dist
+import pyro.poutine as poutine
+from pyro.infer import EmpiricalMarginal
+from pyro.infer.mcmc import MCMC, NUTS
+
+
+def model(sigma):
+    eta_dist = dist.Normal(torch.zeros(data.J), torch.ones(data.J))
+    mu_dist = dist.Normal(loc=torch.zeros(1), scale=10 * torch.ones(1))
+    tau_dist = dist.HalfCauchy(loc=torch.zeros(1), scale=25 * torch.ones(1))
+
+    eta = pyro.sample('eta', eta_dist)
+    mu = pyro.sample('mu', mu_dist)
+    tau = pyro.sample('tau', tau_dist)
+
+    theta = mu + tau * eta
+
+    return pyro.sample("obs", dist.Normal(theta, sigma))
+
+
+def conditioned_model(model, sigma, y):
+    return poutine.condition(model, data={"obs": y})(sigma)
+
+
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+
+parser = argparse.ArgumentParser(description='Eight Schools MCMC')
+parser.add_argument('--num-samples', type=int, default=1000, metavar='NUMSAMPLES',
+                    help='number of MCMC samples (default: 1000)')
+parser.add_argument('--warmup-steps', type=int, default=1000, metavar='WARMUPSTEPS',
```

Comment:
i recommend putting all of this in a `main()` method, see [vae](https://github.com/uber/pyro/blob/dev/examples/vae/vae.py#L197) for example